### PR TITLE
Rename shared configs by grouping in folders

### LINKED ISSUE #525: Reconsider shared config naming scheme
We have an inconsistent naming scheme for "sub configs".

For language rules, we use a path separator:
* language/es5
* language/es6
* _etc._

Elsewhere we are using a hypen, with the rules stored at the root level
* vue-es5
* vue-es6
* mediawiki-qunit
* _etc._

We should pick a consistent scheme.

### PR DESCRIPTION
**Pull request by edg2s**
e.g.
* client-es5 -> client/es5
* mediawiki-quint -> mediawiki/qunit
* vue3-common -> vue3/common

Fixes #525

### CODE REVIEW

**Comment by edg2s** [comment]
This is obviously a large breaking change, so we should be sure this is a sensible approach before merging.

**Comment by jdforrester** [comment]
Can we leave the old names as redirects (`'client-es5': require('client/es5')` _etc._)?

**Comment by edg2s** [comment]
> Can we leave the old names as redirects (`'client-es5': require('client/es5')` _etc._)?

All the old names are kept as aliases for now: https://github.com/wikimedia/eslint-config-wikimedia/pull/527/files#diff-4e1706d3d1da0ad393b3151e9020e0d075c13b930439acc3c6d12ece63ad7da7

**Review by jdforrester — DISMISSED** [review]

**Comment by jdforrester** [comment]
Leaving for review by others before merging, as Ed says.

**Comment by jdforrester** [comment]
Do we want to land this and make a new release?

File: package.json [hunk 1]
```diff
@@ -46,12 +46,16 @@
...
-		"vue2-es5.js",
-		"vue2-es6.js",
-		"vue3-es6.js",
-		"vue-wrappers.js",
+		"vue2-es5.json",
+		"vue2-es6.json",
+		"vue3-es6.json",
 		"yaml.json",
-		"language"
+		"client",
```
Comment (Krinkle) [inline package.json]: If supported, I suggest including a trailing slash to clarify that these are directories. You can use `npm pack` to dry-run a release and see the listed files in the command output (or the tarball file itself).

Comment (edg2s) [reply package.json]: Done

**Comment by edg2s** [comment]
> Do we want to land this and make a new release?

Happy to land it.

**Review by jdforrester — APPROVED** [review]

**Pull request merged**